R/utilities-data-validation.R

```diff
@@ -151,6 +151,45 @@ checkEpsilon <- function(epsilon) {
 }
 
 
+#' Check delta parameter
+#' 
+#' This method is to send the user a warning message if they entered a delta value
+#' that will not be used.
+#' 
+#' If the mechanism is NOT a mechanism that uses a delta value and the user entered a delta value, 
+#' send a warning message saying the delta value with not be used and set the delta 
+#' value to 0. If a mechanism that uses a delta value is being used and the user entered a 
+#' delta value, set it as the delta value (the value of delta will be checked in the 
+#' mechanism). If a mechanism that uses a delta value is being used and the user did 
+#' not enter a delta value, set the delta to the default value (2^-30).
+#' 
+#' @param mechanism The mechanism chosen by determineMechanism
+#' @param delta The delta value entered by the user, may be NULL
+#' 
+#' @return The value that will be used as the delta value for the statistic
+
+checkDelta <- function(mechanism, delta=NULL) {
+    # If a mechanism that uses a delta value is NOT being used, set delta to 0.
+    # If a mechanism that uses a delta value is being used, return the entered value.
+    # If the user did not enter a value, return the default value.
+    # Throw an error if the user entered a value that will not be used.
+    if (!(mechanism %in% c('mechanismStability', 'mechanismGaussian'))) {
```

Comment:
My apologies if I'm misunderstanding something, but should `mechanismExponential` also be included here? It is listed in the comment below (line 182) and in the [testthat utilities file](https://github.com/privacytoolsproject/PSI-Library/pull/103/files#diff-63031389bd8961ee4aaed59a8ff5b06b) as a mechanism with delta != 0.

Reply:
The exponential mechanism as defined in Dwork and Roth preserves (epsilon,0) DP, so in theory it shouldn't have a delta in it. Our implementation does use a delta in the calculation of the gap used. But we also think that this is maybe incorrect, or at least we did last time I talked to Adam/James about it. So, Megan and I decided it's ok to remove delta in this case. It's all a bit of a moot point since the exponential mechanism is broken though. 

We should try to be consistent at least within the commenting of the function and the testthat functions. (Megan, maybe run testthat stuff again and make sure everything runs as expected; one of them fails for me)

Reply:
> The exponential mechanism as defined in Dwork and Roth preserves (epsilon,0) DP, so in theory it shouldn't have a delta in it. Our implementation does use a delta in the calculation of the gap used. But we also think that this is maybe incorrect, or at least we did last time I talked to Adam/James about it. So, Megan and I decided it's ok to remove delta in this case. It's all a bit of a moot point since the exponential mechanism is broken though.
> 
> We should try to be consistent at least within the commenting of the function and the testthat functions. (Megan, maybe run testthat stuff again and make sure everything runs as expected; one of them fails for me)

Yep, you were right, one of the tests still treated the exponential mechanism as a mechanism that uses delta. I have updated the test to fix it.